packages/next/src/client/components/client-page.tsx

```diff
@@ -15,16 +21,33 @@ import type { Params } from '../../server/request/params'
  */
 export function ClientPageRoot({
   Component,
-  searchParams,
-  params,
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  promises,
+  serverProvidedParams,
 }: {
   Component: React.ComponentType<any>
-  searchParams: ParsedUrlQuery
-  params: Params
-  promises?: Array<Promise<any>>
+  serverProvidedParams: null | {
+    searchParams: ParsedUrlQuery
+    params: Params
+    promises: Array<Promise<any>> | null
+  }
 }) {
+  let searchParams: ParsedUrlQuery
+  let params: Params
+  if (serverProvidedParams !== null) {
+    searchParams = serverProvidedParams.searchParams
+    params = serverProvidedParams.params
+  } else {
+    // When clientParamParsing is enabled, the server does not pass the params
```

Comment:
nit: we don't have a distinct flag for this anymore, so will be easier to associate this with cacheComponents

```suggestion
    // When cacheComponents is enabled, the server does not pass the params
```